activity.py

```diff
@@ -133,6 +141,13 @@ def build_toolbar(self):
 
         return toolbar_box
 
+    def _add_hole(self, button):
```

Comment:
Here self.game.hole could be assigned directly from button.get_active().  Once assigned, the next call to get_active() can be avoided.

Reply:
i have refactored the code accordingly